var/spack/repos/builtin/packages/py-pandas/package.py

```diff
@@ -129,6 +131,10 @@ class PyPandas(PythonPackage):
     # Optional dependencies
     # https://pandas.pydata.org/pandas-docs/stable/getting_started/install.html#optional-dependencies
 
+    # https://github.com/spack/spack/issues/40452
+    depends_on("py-openpyxl", type=("run"), when="+excel")
+    depends_on("py-openpyxl@3.0.7:", type=("run"), when="@1.5.3: +excel")
```

Comment:
Missing 4 other deps

Reply:
I must have missed the mentioning of "5" in your previous comment. I need to find out what the other four are.

Reply:
For 2.1.1 it's:
```toml
excel = ['odfpy>=1.4.1', 'openpyxl>=3.0.10', 'pyxlsb>=1.0.9', 'xlrd>=2.0.1', 'xlsxwriter>=3.0.3']
```
You'll have to check older versions too. If you're new to Python packaging, I would suggest reading: https://spack.readthedocs.io/en/latest/build_systems/pythonpackage.html

Reply:
I've done a few Python packages in the past, but thanks. I checked the pandas releases 1.4, 1.5, 2.0, 2.1 and defined those optional dependencies for Excel in the package. I defined the variant `excel` only for those versions of `py-pandas`, because it's likely a waste of my time to code this up for ancient versions. I checked the five dependencies, one didn't exist yet (`py-pyxlsb`), which I added, and one needed additional versions to satisfy the requirements (`py-xlsxwriter`). I am going to update the PR description accordingly.

I tested this for version 1.5.3 on my macOS, and it all installed fine.